app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -57,39 +57,51 @@ class RegistrationSubmissionController @Inject() (
       registration   <- registrationService.getRegistration(id).asResponseError
       additionalInfo <- registrationAdditionalInfoService.get(registration.internalId).asResponseError
       response       <- if (registration.isRegistration) {
-                          registerForEcl(registration, additionalInfo.liabilityYear)
+                          registerForEcl(registration, Some(additionalInfo))
                         } else {
                           subscribeToEcl(registration, additionalInfo)
                         }
     } yield response).convertToResult(OK)
   }
 
-  private def registerForEcl(registration: Registration, liabilityYear: Option[Int])(implicit
+  private def registerForEcl(
+    registration: Registration,
+    registrationAdditionalInfo: Option[RegistrationAdditionalInfo]
+  )(implicit
     hc: HeaderCarrier,
     request: AuthorisedRequest[_]
   ): EitherT[Future, ResponseError, CreateEclSubscriptionResponsePayload] =
     for {
-      _        <- registrationValidationService.validateRegistration(registration).asResponseError
-      now       = Instant.now().truncatedTo(ChronoUnit.SECONDS)
-      response <- dmsService
-                    .submitToDms(registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml), now)
-                    .asResponseError
-      _         = if (appConfig.amendRegistrationNrsEnabled) {
-                    nrsService.submitToNrs(
-                      registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
-                      response.eclReference,
-                      appConfig.eclAmendRegistrationNotableEvent
-                    )
-                  }
-      _         = auditService.successfulSubscriptionAndEnrolment(registration, response.eclReference, liabilityYear)
+      _              <- registrationValidationService
+                          .validateRegistration(EclRegistrationModel(registration, registrationAdditionalInfo))
+                          .asResponseError
+      now             = Instant.now().truncatedTo(ChronoUnit.SECONDS)
+      response       <- dmsService
+                          .submitToDms(registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml), now)
+                          .asResponseError
+      _               = if (appConfig.amendRegistrationNrsEnabled) {
+                          nrsService.submitToNrs(
+                            registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+                            response.eclReference,
+                            appConfig.eclAmendRegistrationNotableEvent
+                          )
+                        }
+      additionalInfo <- valueOrError(registrationAdditionalInfo, "")
```

Comment:
Please add a description here rather than empty string, in this case it would be "Registration additional info"

Reply:
Oh, good catch, forgot to add this one 😄

Reply:
Done :)